Create roles.md (#644)
vtex-apps/io-documentation: docs/en/Concepts/roles.md

# Roles

> Roles are uniquely identifiable names given to the actors of a request. This means that, in VTEX IO, whenever a request is performed, the actor of this action receives a role. In the sequence, to determine whether or not that call can go through, the role is translated into a set of policies.
> 

Reviewer: ```suggestion
Roles are uniquely identifiable names given to request sources. 

Whenever a request is performed in the VTEX IO platform, the source of this action, meaning the user or app responsible for performing the request, automatically receives a unique role.

To detect whether or not a user is authorized to perform a request, and consequently to determine whether or not a request can go through the platform, this role is interpreted by the VTEX IO platform who checks if it has the needed [policies]() attached.
```

Reviewer: não sei se seria who checks ou that checks kkkk

Author: ```suggestion
Roles are uniquely identifiable names given to request sources. 

Whenever a request is performed in the VTEX IO platform, the source of this action, which can be the user or app responsible for performing that request, automatically receives a unique role.

In the sequence, to detect whether or not the request source is authorized to perform that particular request, and also to determine if the call can go through, VTEX IO platform checks if the source role contains the required policies attached.
```

> Since the same user can have different permissions within the same account, each role can have multiple policies attached.

Reviewer: ```suggestion
<div class="alert alert-info">
Since a single user can have different permissions within the same account, each role can also have multiple policies attached to it.</div>
```

> A role can be given to a user or an app. The general rule is to have an identity provider, such as `user`, `app`, or `link`, followed by an identity.

Reviewer: ```suggestion
Since apps and users are able to perform a request on the platform, both of them mandatorily receive roles. However, notice the following: app and user roles are structured in different formats.
```

> For example, if a user makes a request, they will usually receive a role like `user:f1sd8705-b34f-48ff-a113-2b8bb8c22411` or `user:<e-mail>`. Or, for example, an app that is installed in an account will get a role such as `app:vtex.auth-server@1.0.0` or, if it is linked, `link:vtex.auth-server@1.0.0`.

Reviewer: ```suggestion
For example: an user making a request would receive a role similar to `user:f1sd8705-b34f-48ff-a113-2b8bb8c22411`, or `user:<e-mail>`. Installed and linked apps, in turn, would receive roles as `app:vtex.auth-server@1.0.0` and `link:vtex.auth-server@1.0.0`.
```

> Aside from the identity types above, there are also support roles, which are a *special* kind of role used only in support apps.

Reviewer: ```suggestion
```

Reviewer: Como Support apps ainda tá meio obscuro, eu tiraria pra não causar ainda mais confusão.

# Roles

Roles are uniquely identifiable names given to request sources. 

Whenever a request is performed in the VTEX IO platform, the source of this action, which can be the user or app responsible for performing that request, automatically receives a unique role.

In the sequence, to detect whether or not the request source is authorized to perform that particular request, and also to determine if the call can go through, VTEX IO platform checks if the source role contains the required policies attached.

> <div class="alert alert-info">
> Since a single request source can have different permissions within the same account, each role can also have multiple policies attached to it.</div>

Author: ```suggestion

<div class="alert alert-info">
As a single request source can have different permissions within the same account, each role can also have multiple policies attached to it.</div>
```

> A role can be given to a user or an app. The general rule is to have an identity provider, such as `user`, `app`, or `link`, followed by an identity.

Author: ```suggestion
Since apps and users are able to perform a request on the platform, both of them mandatorily receive roles. However, notice the following: app and user roles are structured in different formats.
```
